renderer: create fence after each surface pass

gulafaran opened the pull request:
instead of adding the buffers to a vector and late release it on the fence in endrender that signals on ALL prior opengl work.

now its making a synckhr after the surface pass. and in endrender duping the native fd , and moving the dup nativefd to a seperate function in ceglsync that takes a param if we want to flush or not. in this case the endrender fence flushes, the buffer fence just gets the fd. and we are releasing buffers individually as early as possible with as few glflush as we can. however still warrants for some benchmarking and testing. 

because it now means creating a eglsynckhr and fd per surface. gain some, loose some.

vaxerski commented:
what about toplevel export?

Zebra2711 commented:
@gulafaran what happened with the old version [87f7a3a](https://github.com/hyprwm/Hyprland/commit/87f7a3a208d4496914bea7184e641d10dbbbef0b)?
adding `m_sync->dupNativeFence(true)` in func `void CMonitorFrameScheduler::onFinishRender` makes my obs record with `render:new_render_scheduling = true` have freeze frames and some screen artifacts

gulafaran commented:
> @gulafaran what happened with the old version [87f7a3a](https://github.com/hyprwm/Hyprland/commit/87f7a3a208d4496914bea7184e641d10dbbbef0b)? adding `m_sync->dupNativeFence(true)` in func `void CMonitorFrameScheduler::onFinishRender` makes my obs record with `render:new_render_scheduling = true` have freeze frames and some screen artifacts

well it merely restored old behaviour, since i moved the duping of the fd into that function, without it on readable would be called on an invalid fd, odd it affected it at all.

gulafaran commented:
technically i think the new_render_sheduling is a bit wrong, i have it disabled myself because it really messes up dual gpu / external monitor. its also by default off if im not mistaken. but i might have missed something obvious with my changes aswell heh. im gonna be offgrid until wednesday tho. so its gonna have to wait for figuring out what or why is happening here

gulafaran commented:
> > technically i think the new_render_sheduling is a bit wrong, i have it disabled myself because it really messes up dual gpu / external monitor. its also by default off if im not mistaken. but i might have missed something obvious with my changes aswell heh. im gonna be offgrid until wednesday tho. so its gonna have to wait for figuring out what or why is happening here
> 
> Yeah, i ran a test with three versions using `render:new_render_scheduling = true`: without this mr, with the old version [87f7a3a](https://github.com/hyprwm/Hyprland/commit/87f7a3a208d4496914bea7184e641d10dbbbef0b) and the latest. It seems the issue happens on all versions. So, I think that weird stuff is mainly caused by `new_render_scheduling`, but it's strange that it happens randomly

reverted a commitpending in dolater, fixed my obs. but im still seeing a weird artifacting on launch of hyprland for a frame or two. otherwise the new render sheduling seems happy here.

gulafaran commented:
> > technically i think the new_render_sheduling is a bit wrong, i have it disabled myself because it really messes up dual gpu / external monitor. its also by default off if im not mistaken. but i might have missed something obvious with my changes aswell heh. im gonna be offgrid until wednesday tho. so its gonna have to wait for figuring out what or why is happening here
> 
> Yeah, i ran a test with three versions using `render:new_render_scheduling = true`: without this mr, with the old version [87f7a3a](https://github.com/hyprwm/Hyprland/commit/87f7a3a208d4496914bea7184e641d10dbbbef0b) and the latest. It seems the issue happens on all versions. So, I think that weird stuff is mainly caused by `new_render_scheduling`, but it's strange that it happens randomly

reverted a commitpending in dolater, fixed my obs. but im still seeing a weird artifacting on launch of hyprland for a frame or two. otherwise the new render sheduling seems happy here.

> > > technically i think the new_render_sheduling is a bit wrong, i have it disabled myself because it really messes up dual gpu / external monitor. its also by default off if im not mistaken. but i might have missed something obvious with my changes aswell heh. im gonna be offgrid until wednesday tho. so its gonna have to wait for figuring out what or why is happening here
> > 
> > 
> > Yeah, i ran a test with three versions using `render:new_render_scheduling = true`: without this mr, with the old version [87f7a3a](https://github.com/hyprwm/Hyprland/commit/87f7a3a208d4496914bea7184e641d10dbbbef0b) and the latest. It seems the issue happens on all versions. So, I think that weird stuff is mainly caused by `new_render_scheduling`, but it's strange that it happens randomly
> 
> reverted a commitpending in dolater, fixed my obs. but im still seeing a weird artifacting on launch of hyprland for a frame or two. otherwise the new render sheduling seems happy here.

ok nvm so it only worked fine on dual monitors, going single it brought back the reason for vaxry adding dolater -_-. time do dig.
